client/lib/support/support-user/localforage-bypass.js

```diff
@@ -0,0 +1,230 @@
+	return promise;
+}
+
+function iterate( callback ) {
+	const promise = new Promise( ( resolve, reject ) => {
+		this.ready().then( () => {
+			try {
+				const db = this._dbInfo.db;
+
+				for ( let key in db ) {
+					const result = db[ key ];
+
+					callback( result, key );
+				}
+
+				resolve();
+			} catch ( e ) {
+				reject( e );
+			}
+		} ).catch( reject );
+	} );
+
+	executeCallback( promise, callback );
+	return promise;
+}
+
+function _key( n, callback ) {
+	const promise = new Promise( ( resolve, reject ) => {
+		this.ready().then( () => {
+			const db = this._dbInfo.db;
+			let result = null;
+			let index = 0;
+
+			for ( let key in db ) {
+				if ( db.hasOwnProperty( key ) && db[ key ] !== undefined ) {
+					if ( n === index ) {
+						result = key;
+						break;
+					}
+					index++;
+				}
+			}
+
+			resolve( result );
+		} ).catch( reject );
+	} );
+
+	executeCallback( promise, callback );
+	return promise;
+}
+
+function _keys( callback ) {
+	const promise = new Promise( ( resolve, reject ) => {
+		this.ready().then( () => {
+			const db = this._dbInfo.db;
+			const keys = [ ];
+
+			for ( let key in db ) {
+				if ( db.hasOwnProperty( key ) ) {
+					keys.push( key );
+				}
+			}
+
+			resolve( keys );
+		} ).catch( reject );
+	} );
+
+	executeCallback( promise, callback );
+	return promise;
+}
+
+function length( callback ) {
+	const promise = new Promise( ( resolve, reject ) => {
+		this.keys().then( ( keys ) => {
+			resolve( keys.length );
+		} ).catch( reject );
+	} );
+
+	executeCallback( promise, callback );
+	return promise;
+}
+
+function removeItem( key, callback ) {
+	debug( 'localForage bypass', 'removeItem', key );
+
+	const promise = new Promise( ( resolve, reject ) => {
+		if ( typeof key !== 'string' ) {
+			return reject();
+		}
+
+		this.ready().then( () => {
+			const db = this._dbInfo.db;
+			if ( db.hasOwnProperty( key ) ) {
+				delete db[ key ];
+			}
+
+			resolve();
+		} ).catch( reject );
+	} );
+
+	executeCallback( promise, callback );
+	return promise;
+}
+
+function setItem( key, value, callback ) {
+	debug( 'localForage bypass', 'setItem', key );
+
+	const promise = new Promise( ( resolve, reject ) => {
+		if ( typeof key !== 'string' ) {
+			return reject();
+		}
+
+		this.ready().then( () => {
+			// Convert undefined values to null.
+			// https://github.com/mozilla/localForage/pull/42
+			if ( value === undefined ) {
+				value = null;
+			}
+
+			// Save the original value to pass to the callback.
+			const originalValue = value;
+
+			let db = this._dbInfo.db;
+			db[ key ] = value;
+			resolve( originalValue );
+		} ).catch( reject );
+	} );
+
+	executeCallback( promise, callback );
+	return promise;
+}
+
+function executeCallback( promise, callback ) {
+	if ( callback ) {
+		promise.then( ( result ) => {
+			callback( null, result );
+		}, ( error ) => {
+			callback( error );
+		} );
+	}
+}
+
+export default {
+	_driver: 'localForageBypass',
+	_initStorage: _initStorage,
```

Comment:
You could alternative declare them as exports at the time of being declared. Not a huge advantage, but might help with clarity of what's being exported.

```js
export const _driver = 'localForageBypass';

export function _initStorage( options ) {
	// ...
}
```